### PR TITLE
Deprecate getMetadata() method, switch test to replacement method.

### PR DESCRIPTION
**Pull request by IanMayo**
Signed-off-by: Ian Mayo <ian@planetmayo.com>

Proposed fix for #30

### CODE REVIEW

File: org.eclipse.january.test/src/org/eclipse/january/metadata/MetadataProviderTest.java [hunk 1]
```diff
@@ -28,14 +28,14 @@ public void testGetMetadata() throws MetadataException {
...
-		Assert.assertTrue(mdp.getMetadata() instanceof IMetadata);
+		Assert.assertTrue(mdp.getFirstMetadata(IMetadata.class) instanceof IMetadata);
 		mdp.clearMetadata(null);
 
 		// add sub-class of IMetadata
 		mdp.addMetadata(new InnerMetadata());
-		System.err.println(mdp.getMetadata());
-		Assert.assertTrue(mdp.getMetadata() instanceof IMetadata);
-		Assert.assertTrue(mdp.getMetadata() instanceof IInnerMetadata);
+		System.err.println(mdp.getFirstMetadata(IInnerMetadata.class));
```
Comment (IanMayo) [inline org.eclipse.january.test/src/org/eclipse/january/metadata/MetadataProviderTest.java]: @PeterC-DLS I'm unsure if the clazz should be ````IMetadata```` or ````IInnerMetadata```` in this test.  Line 37 fails if ````IMetadata```` is used as clazz parameter.

Comment (PeterC-DLS) [inline org.eclipse.january.test/src/org/eclipse/january/metadata/MetadataProviderTest.java]: Yeah, it's going to be a little more complicated. Historical metadata users have sub-classed `IMetadata` which is not very scalable and, now, discouraged. As the test implies, they expect both asserts to be true thus the input class can be `IMetadata` or `IInnerMetadata` and return the same thing in this case. The solution is to modify the behaviour of `addMetadata()` and `setMetadata()` (may be just for the `IMetadata` case)

**Comment by PeterC-DLS** [comment]
Please see my alternative in #33

**Comment by IanMayo** [comment]
Closing, superceded by #33

**Pull request closed without merging**